test/ibmq/test_ibmq_logger.py

```diff
@@ -0,0 +1,177 @@
+        """Test setting up a logger with valid log levels, case insensitive."""
+        level_to_set = logging.INFO
+        valid_log_levels = ['INFO', 'info', 'InFo']
+
+        for valid_log_level in valid_log_levels:
+            with patch.dict('os.environ', {self.log_level_env: valid_log_level}):
+                setup_logger(self.logger)
+                self.assertEqual(self.logger.level, level_to_set,
+                                 'The logger level was set to {}, but it should '
+                                 'be {}.'.format(self.logger.level, logging.INFO))
+
+    def test_valid_log_levels(self):
+        """Test setting up a logger with all valid levels, case insensitive."""
+        all_valid_log_levels = {
+            'debug': logging.DEBUG, 'iNFo': logging.INFO,
+            'WARNING': logging.WARNING, 'error': logging.ERROR,
+            'CRITICAL': logging.CRITICAL}
+
+        for level_name, level_value in all_valid_log_levels.items():
+            with patch.dict('os.environ', {self.log_level_env: level_name}):
+                setup_logger(self.logger)
+                self.assertEqual(self.logger.level, level_value,
+                                 'The logger level was set to {}, but it should '
+                                 'be {}.'.format(self.logger.level, level_value))
+
+    def test_log_file(self):
+        """Test setting up a logger by specifying a file and log level."""
+        log_level_warning = ('WARNING', logging.WARNING)
+        env_vars_to_patch = {self.log_level_env: log_level_warning[0],
+                             self.log_file_env: self.log_file}
+
+        with patch.dict('os.environ', env_vars_to_patch):
+            setup_logger(self.logger)
+            self.assertEqual(self.logger.level, log_level_warning[1],
+                             'The logger level was set to {}, but it should '
+                             'be {}.'.format(self.logger.level, log_level_warning[1]))
+
+            # Assert the file handler was created.
+            self.assertTrue(self.logger.handlers,
+                            'A file handler should have been setup, but it was not.')
+            self.assertEqual(len(self.logger.handlers), 1,
+                             'Many handlers were setup {}, but it should have only '
+                             'been one.'.format(self.logger.handlers))
+
+            # Note that only messages >= `WARNING` will be logged.
+            self.logger.debug('This is a debug message that should not be logged in the file.')
+            self.logger.warning('This is a warning message that should be logged in the file.')
+            self.logger.error('This is a error message that should be logged in the file.')
+
+            # Retrieve the file handler.
+            file_handler = self.logger.handlers[0]
+            file_path = file_handler.baseFilename
+
+            # Assert the file exists.
+            self.assertTrue(os.path.exists(file_path),
```

Comment:
the `file_path` used for verification should be whatever you passed into setup_logger (i.e. self.log_file) and not from the handler.